test/integration/apiserver/crd_validation_expressions_test.go

```diff
@@ -301,6 +301,36 @@ func TestCustomResourceValidators(t *testing.T) {
 			t.Error("Unexpected error creating custom resource but metadata validation rule")
 		}
 	})
+	t.Run("CR creation MUST fail if a x-kubernetes-validations rule exceeds the runtime cost limit", func(t *testing.T) {
+		structuralWithValidators := crdWithSchema(t, "RuntimeCostLimit", structuralSchemaWithCostLimit)
+		crd, err := fixtures.CreateNewV1CustomResourceDefinition(structuralWithValidators, apiExtensionClient, dynamicClient)
+		if err != nil {
+			t.Errorf("Unexpected error creating custom resource definition: %v", err)
+		}
+		gvr := schema.GroupVersionResource{
+			Group:    crd.Spec.Group,
+			Version:  crd.Spec.Versions[0].Name,
+			Resource: crd.Spec.Names.Plural,
+		}
+		crClient := dynamicClient.Resource(gvr)
+		name1 := names.SimpleNameGenerator.GenerateName("cr-1")
+		cr := &unstructured.Unstructured{Object: map[string]interface{}{
+			"apiVersion": gvr.Group + "/" + gvr.Version,
+			"kind":       crd.Spec.Names.Kind,
+			"metadata": map[string]interface{}{
+				"name": name1,
+			},
+			"spec": map[string]interface{}{
+				"list": genLargeArray(725, 20),
+			},
+		}}
+		_, err = crClient.Create(context.TODO(), cr, metav1.CreateOptions{})
```

Comment:
non-blocking, but I'm curious how long this create call takes before it fails (to see if our runtime protections are actually keeping the request handling time short)

Reply:
For what it's worth, averaged over 5 runs on my machine it takes about 539ms before it fails.